spec/Section 6 -- Execution.md

```diff
@@ -322,39 +312,315 @@ Unsubscribe(responseStream):
+  - If {data} on {result} is {null}:
+    - Initialize {completed} to an empty list.
+    - Let {parents} be the parent nodes of {executionGroup}.
+    - Initialize {completed} to an empty list.
+    - For each {pendingResult} of {parents}:
+      - Append {GetCompletedEntry(parent, errors)} to {completed}.
+      - Remove {pendingResult} and all of its descendant nodes from {graph},
+        except for any descendant Incremental Data Record nodes with other
+        parents.
+    - Let {hasNext} be {false}, if {graph} is empty.
+    - Yield an unordered map containing {completed} and {hasNext}.
+    - Continue to the next completed Pending Incremental Data node.
+  - Replace {node} in {graph} with a new node corresponding to the Completed
+    Incremental Data for {result}.
+  - Let {resultIncrementalDataRecords} be {incrementalDataRecords} on {result}.
+  - Update {graph} to {GraphFromRecords(resultIncrementalDataRecords, graph)}.
+  - Let {completedDeferredFragments} be the set of root nodes in {graph} without
+    any child Pending Data nodes.
+  - Let {completedIncrementalDataNodes} be the set of completed Incremental Data
+    nodes that are children of {completedDeferredFragments}.
+  - If {completedIncrementalDataNodes} is empty, continue to the next completed
+    Pending Incremental Data Node.
+  - Initialize {incremental} to an empty list.
+  - For each {node} of {completedIncrementalDataNodes}:
+    - Let {incrementalDataRecord} be the corresponding record for {node}.
+    - Append {GetIncrementalEntry(incrementalDataRecord, graph)} to
+      {incremental}.
+    - Remove {node} from {graph}.
+  - Initialize {completed} to an empty list.
+  - For each {pendingResult} of {completedDeferredFragments}:
+    - Append {GetCompletedEntry(pendingResult)} to {completed}.
+    - Remove {pendingResult} from {graph}, promoting its child Deferred Fragment
+      nodes to root nodes.
+  - Let {newPendingResults} be the result of {GetNonEmptyNewPending(graph)}.
+  - Add all nodes in {newPendingResults} to {pendingResults}.
+  - Update {graph} to the subgraph rooted at nodes in {pendingResults}.
+  - Let {pending} be the result of {GetPendingEntry(newPendingResults)}.
+  - Yield the result of {GetIncrementalResult(graph, incremental, completed,
+    pending)}.
+- Complete this incremental result stream.
+
+GraphFromRecords(incrementalDataRecords, graph):
+
+- Let {newGraph} be a new directed acyclic graph containing all of the nodes and
+  edges in {graph}.
+- For each {incrementalDataRecord} of {incrementalDataRecords}:
+  - Add {incrementalDataRecord} to {newGraph} as a new Pending Data node
+    directed from the {pendingResults} that it completes, adding each of
+    {pendingResults} to {newGraph} as a new node directed from its {parent},
+    recursively adding each {parent} until {incrementalDataRecord} is connected
+    to {newGraph}, or the {parent} is not defined.
+- Return {newGraph}.
+
+GetNonEmptyNewPending(graph):
```

Comment:
rename to match js implementation? getNewRootNodes?

Reply:
359441e